activestorage/app/models/active_storage/blob/representable.rb

```diff
@@ -112,10 +112,10 @@ def default_variant_format
     end
 
     def format
-      if filename.extension.present? && MiniMime.lookup_by_extension(filename.extension)&.content_type == content_type
+      if filename.extension.present? && Marcel::MimeType.for(extension: filename.extension) == content_type
         filename.extension
       else
-        MiniMime.lookup_by_content_type(content_type)&.extension
+        Marcel::Magic.new(content_type.to_s).extensions.first
```

Comment:
Makes sure `content_type` is not nil, which can raise if passed to `Marcel::Magic.new`